pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -211,6 +212,10 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable SqlNodeAndOption
             .putIfAbsent(Broker.Request.QueryOptionKey.ENABLE_NULL_HANDLING, _enableNullHandling);
       }
 
+      sqlNodeAndOptions.getOptions().putIfAbsent(QueryOptionKey.REGEXP_LIKE_DICTIONARY_THRESHOLD, String.valueOf(
```

Comment:
To reduce overhead, let's put it only when the value is explicitly set on broker